src/manifest/pythonEnvironment.ts

```diff
@@ -118,50 +153,61 @@ export class PythonEnvironment implements Disposable {
+        }
+        try {
+          const integratedEnv:
+            | Record<string, Record<string, string>>
+            | undefined = workspace
+            .getConfiguration("terminal")
+            .get("integrated.env");
+          if (integratedEnv) {
+            // parse vs code environment variables
+            for (const prop in integratedEnv) {
+              // Ignore any settings not supported by the terminal
+              // We don't know which os is used in terminal unfortunately, so we just merge all of them.
+              if (!["osx", "windows", "linux"].includes(prop)) {
+                this.dbtTerminal.debug(
+                  "pythonEnvironment:envVars",
+                  "Loading env vars from config.terminal.integrated.env",
+                  "Ignoring invalid property " + prop,
+                );
+                continue;
+              }
               this.dbtTerminal.debug(
-                "pythonEnvironment",
+                "pythonEnvironment:envVars",
                 "Loading env vars from config.terminal.integrated.env",
-                "Ignoring invalid property  " + prop,
+                "Merging from " + prop,
+                Object.keys(integratedEnv[prop]),
               );
-              continue;
+              for (const key in integratedEnv[prop]) {
+                envVars[key] = this.substituteSettingsVariables(
+                  integratedEnv[prop][key],
+                  process.env,
+                );
+                this.environmentVariableSource[key] = "integrated";
+              }
             }
-            const vsCodeEnv = env[prop];
-            const newEnvVars = this.parseEnvVarsFromUserSettings(vsCodeEnv);
-            this.dbtTerminal.debug(
-              "pythonEnvironment",
-              "Loading env vars from config.terminal.integrated.env",
-              "Merging from " + prop,
-              newEnvVars,
-            );
-            envVars = {
-              ...process.env,
-              ...envVars,
-              ...newEnvVars,
-            };
           }
-        }
-        try {
           if (api.environment) {
-            envVars = {
-              ...envVars,
-              ...api.environments.getEnvironmentVariables(
-                workspace.workspaceFolders![0],
-              ),
-            };
+            const workspacePath = workspace.workspaceFolders![0];
+            this.dbtTerminal.debug(
+              "pythonEnvironment:envVars",
+              `workspacePath:${workspacePath.uri.fsPath}`,
+            );
+            const workspaceEnv =
+              api.environments.getEnvironmentVariables(workspacePath);
+            this.dbtTerminal.debug(
+              "pythonEnvironment:envVars",
+              `workspaceEnv:${Object.keys(workspaceEnv)}`,
+            );
+            for (const key in workspaceEnv) {
+              envVars[key] = workspaceEnv[key];
+              this.environmentVariableSource[key] = "dotenv";
```

Comment:
A lot of env variables from process are actually reported as coming from dotenv.